### PR TITLE
Consolidate context request endpoints into a single request endpoint

### PR DESCRIPTION
**Pull request by lewis-sanchez**
This PR consolidates the previous two requests to get and generate server contextualization into a single get context request.

### CODE REVIEW

File: src/Microsoft.SqlTools.ServiceLayer/Metadata/MetadataService.cs [hunk 1]
```diff
@@ -59,7 +59,6 @@ public void InitializeService(ServiceHost serviceHost)
             serviceHost.SetRequestHandler(MetadataListRequest.Type, HandleMetadataListRequest, true);
             serviceHost.SetRequestHandler(TableMetadataRequest.Type, HandleGetTableRequest, true);
             serviceHost.SetRequestHandler(ViewMetadataRequest.Type, HandleGetViewRequest, true);
-            serviceHost.SetRequestHandler(GenerateServerContextualizationRequest.Type, HandleGenerateServerContextualizationNotification, true);
```
Comment (lewis-sanchez) [inline src/Microsoft.SqlTools.ServiceLayer/Metadata/MetadataService.cs:62]: Wasn't entirely sure if I should keep the original 2 endpoints and add a 3rd one that handles both getting and generating context. I can add it back, but I also don't see a need for 3 endpoints (get context, generate context, get and generate context)

Comment (Charles-Gagnon) [reply src/Microsoft.SqlTools.ServiceLayer/Metadata/MetadataService.cs:62]: Just having the one makes sense to me!

File: src/Microsoft.SqlTools.ServiceLayer/Metadata/MetadataService.cs [hunk 2]
```diff
@@ -138,116 +134,64 @@ public void InitializeService(ServiceHost serviceHost)
...
-        internal static async Task GenerateServerContextualization(GenerateServerContextualizationParams contextualizationParams, RequestContext<GenerateServerContextualizationResult> requestContext)
+        internal static async Task GetServerContextualization(GetServerContextualizationParams contextualizationParams, RequestContext<GetServerContextualizationResult> requestContext)
         {
             MetadataService.ConnectionServiceInstance.TryFindConnection(contextualizationParams.OwnerUri, out ConnectionInfo connectionInfo);
-
-            if (connectionInfo != null)
+            if (connectionInfo == null)
+            {
+                Logger.Error("Failed to find connection info about the server.");
+                await requestContext.SendError("Failed to find connection info about the server.");
```
Comment (Charles-Gagnon) [inline src/Microsoft.SqlTools.ServiceLayer/Metadata/MetadataService.cs:143]: I'd suggest just throwing an exception. That way you're guaranteeing it won't keep executing and makes it clear that that's the end of execution

(STS will catch exceptions and send them as errors by default)

File: src/Microsoft.SqlTools.ServiceLayer/Metadata/MetadataService.cs [hunk 3]
```diff
@@ -138,116 +134,64 @@ public void InitializeService(ServiceHost serviceHost)
...
+            }
+            else
             {
-                using (SqlConnection sqlConn = ConnectionService.OpenSqlConnection(connectionInfo, "metadata"))
+                if (connectionInfo != null)
```
Comment (Charles-Gagnon) [inline src/Microsoft.SqlTools.ServiceLayer/Metadata/MetadataService.cs:147]: Not needed

File: src/Microsoft.SqlTools.ServiceLayer/Metadata/MetadataService.cs [hunk 4]
```diff
@@ -138,116 +134,64 @@ public void InitializeService(ServiceHost serviceHost)
...
+                                    await requestContext.SendError(ex);
+                                }
                             }
-                            catch (Exception ex)
+                            else
                             {
-                                Logger.Error($"An error was encountered while writing server contextualization scripts to the cache. Error: {ex.Message}");
-                                await requestContext.SendError(ex);
+                                Logger.Error("Failed to generate server contextualization scripts");
+                                await requestContext.SendError(SR.FailedToGenerateServerContextualizationScripts);
```
Comment (Charles-Gagnon) [inline src/Microsoft.SqlTools.ServiceLayer/Metadata/MetadataService.cs:174]: throw exception here too

File: src/Microsoft.SqlTools.ServiceLayer/Metadata/MetadataService.cs [hunk 5]
```diff
@@ -138,116 +134,64 @@ public void InitializeService(ServiceHost serviceHost)
...
+                                    {
+                                        Context = string.Join('\n', scripts)
+                                    });
+
+                                    MetadataScriptTempFileStream.Write(connectionInfo.ConnectionDetails.ServerName, scripts);
+                                }
+                                catch (Exception ex)
+                                {
+                                    Logger.Error($"An error was encountered while writing server contextualization scripts to the cache. Error: {ex.Message}");
+                                    await requestContext.SendError(ex);
```
Comment (Charles-Gagnon) [inline src/Microsoft.SqlTools.ServiceLayer/Metadata/MetadataService.cs:168]:
```suggestion
                                    throw;
```

File: src/Microsoft.SqlTools.ServiceLayer/Metadata/MetadataService.cs [hunk 6]
```diff
@@ -138,116 +134,64 @@ public void InitializeService(ServiceHost serviceHost)
...
+                                    await requestContext.SendResult(new GetServerContextualizationResult()
+                                    {
+                                        Context = string.Join('\n', scripts)
+                                    });
+
+                                    MetadataScriptTempFileStream.Write(connectionInfo.ConnectionDetails.ServerName, scripts);
+                                }
+                                catch (Exception ex)
+                                {
+                                    Logger.Error($"An error was encountered while writing server contextualization scripts to the cache. Error: {ex.Message}");
```
Comment (Charles-Gagnon) [inline src/Microsoft.SqlTools.ServiceLayer/Metadata/MetadataService.cs:167]: This isn't necessarily true, maybe just make this generic "while generating server contextualization scripts"

File: src/Microsoft.SqlTools.ServiceLayer/Metadata/MetadataService.cs [hunk 7]
```diff
@@ -138,116 +134,64 @@ public void InitializeService(ServiceHost serviceHost)
...
                         {
-                            Context = string.Join('\n', scripts)
-                        });
-                    }
-                    catch (Exception ex)
-                    {
-                        Logger.Error("Failed to read scripts from the script cache");
-                        await requestContext.SendError(ex);
+                            Logger.Error("Failed to read scripts from the script cache");
+                            await requestContext.SendError(ex);
```
Comment (Charles-Gagnon) [inline src/Microsoft.SqlTools.ServiceLayer/Metadata/MetadataService.cs:191]:
```suggestion
                            throw;
```

**Review by Charles-Gagnon — APPROVED** [review]

File: src/Microsoft.SqlTools.ServiceLayer/Metadata/Contracts/GetServerContextualizationRequest.cs [hunk 8]
```diff
@@ -10,22 +10,22 @@ namespace Microsoft.SqlTools.ServiceLayer.Metadata.Contracts
...
-        /// The generated server context.
+        /// The generated context.
         /// </summary>
-        public string? Context { get; set; }
+        public string Context { get; set; }
     }
 
     public class GetServerContextualizationRequest
     {
-        public static readonly RequestType<GetServerContextualizationParams, GetServerContextualizationResult> Type =
```
Comment (kisantia) [inline src/Microsoft.SqlTools.ServiceLayer/Metadata/Contracts/GetServerContextualizationRequest.cs:28]: why did the readonly need to be removed?

File: src/Microsoft.SqlTools.ServiceLayer/Metadata/MetadataService.cs [hunk 9]
```diff
@@ -121,15 +120,12 @@ public void InitializeService(ServiceHost serviceHost)
             await HandleGetTableOrViewRequest(metadataParams, "view", requestContext);
         }
 
-        /// <summary>
-        /// Handles the event for generating server contextualization scripts.
-        /// </summary>
-        internal static Task HandleGenerateServerContextualizationNotification(GenerateServerContextualizationParams contextualizationParams,
-            RequestContext<GenerateServerContextualizationResult> requestContext)
+        internal static Task HandleGetServerContextualizationRequest(GetServerContextualizationParams contextualizationParams,
```
Comment (kisantia) [inline src/Microsoft.SqlTools.ServiceLayer/Metadata/MetadataService.cs:123]: please add doc comments

**Review by kisantia — APPROVED** [review]

**Pull request merged**